src/freva/_history.py

```diff
@@ -25,6 +26,7 @@ def history(
     return_results: bool = False,
     return_command: bool = False,
     _return_dict: bool = True,
+    all_users: bool = False,
```

Comment:
Why don't you make this a user flag, that takes None as default and if it is none then the current user is taken. If `all` all users are taken, if any other user any other user is taken?

Reply:
I did this to simplify, mirroring what is happening in the web side. But I will take a look

Reply:
I am seeing that if I want to add that functionality I will probably need to modify something more fundamental in the [User class](https://github.com/FREVA-CLINT/freva/blob/main/src/evaluation_system/model/user.py)

for example, if I take the simplest case and I create previously a user:

```
...
from django.contrib.auth.models import User
from evaluation_system.model.history.models import History
N = 10
random_string = "".join(random.choices(string.ascii_lowercase + string.digits, k=N))
other_user = User.objects.create_user(
    username=f"user_{random_string}", password="123"
)
History.objects.create(
    timestamp=datetime.datetime.now(),
    status=History.processStatus.running,
    uid=other_user,
    configuration='{"some": "config", "dict": "values"}',
    tool=f"dummytool_{random_string}",
    slurm_output="/path/to/slurm-44742.out",
    host=socket.gethostbyname(socket.gethostname()),
)
```
that will not be enough to then search for it, specifically, since this is a django user type and we need a evaluation user type:

```
from evaluation_system.model.user import User

User(other_user.username)

Cell In[5], line 1
----> 1 User(other_user.username)

File [~/work/freva/github/freva/src/evaluation_system/model/user.py:64](http://localhost:8888/home/etor/work/freva/github/freva/src/evaluation_system/model/user.py#line=63), in User.__init__(self, uid, email)
     62 print(f"User class __init__ {self._userdata = }")
     63 if isinstance(uid, str):
---> 64     self._userdata = pwd.getpwnam(uid)
     65 else:
     66     self._userdata = pwd.getpwuid(uid)

KeyError: "getpwnam(): name not found: 'user_mnwewlipdo'"
```

For that I need to somewhat create a user, but I don't see how with the evaluation User class, so I am using the DummyUser for it:
```
from evaluation_system.tests.mocks.dummy import DummyUser
temp_user = DummyUser(random_home=True, pw_name=f"user_{random_string}", pw_passwd="123")
temp_user._userdata = pwd.struct_passwd(temp_user._userdata)
```
But when calling it via
```
User(temp_user.getName())
```
I get the same error as with the Django User


any ideas?

Reply:
@eelucio i think these two models have two different stories. First when you define a user via django User class:
```python
from django.contrib.auth.models import User
N = 10
random_string = "".join(random.choices(string.ascii_lowercase + string.digits, k=N))
other_user = User.objects.create_user(
    username=f"user_{random_string}", password="123"
)
```
Then you can use it in History model to create a new object, because `uid` of History has a django User type. https://github.com/FREVA-CLINT/freva/blob/b2fd2b751ec1999fdfbfc2731d9354f75493c463/src/evaluation_system/model/history/models.py#L98 . Otherwise you cannot use it to make it with evaluation_system User model.

Now you made your django User type and History object. It's done here.

In the next level, you imported evaluation system User model and i think you want to convert the django User type `temp_user` to evaluation_system User type which I don't know why and it's not possible. and the difference between these two stories is here:

If you look at the evaluation_system User class, it uses `pwd` to fetch the users from the password database which is located in `/etc/passwd`. it means users of ur machine. if you do the following you will get the all users of your machine. 
```bash
>>> import pwd
>>> pwd.getpwall()
```
So, it means when you make a username via django, it makes it in your db, but evaluation_system just take advantage of your system's usernames. and if you want to make it in your machine to call it via evaluation_system User model, you have to add it like here. 

```bash
sudo useradd temp_user._userdata
```
But it needs to be unique. And the main point I really couldn't understand what your purpose is to convert django User model to evalution_system User model? What you want to achieve with this?

Reply:
Thanks @MoSHad91 ,

if I want to use a `--user="test_user"` flag, this user needs to be searched in the database, so I either do it directly in [get_history()](https://github.com/FREVA-CLINT/freva/blob/main/src/evaluation_system/api/plugin_manager.py#L1010):
```python

    return user_.getUserDB().getHistory(
        tool_name=plugin_name.lower() if plugin_name else None,
        limit=limit,
        since=since,
        until=until,
        entry_ids=entry_ids,
        uid="test_user", <--- # here is the str of the username
    )
```
or I need to pass it as `User("test_user")` to then get

```python
uid=user.getName()
```
which would be my approach. I would guess that passing a string directly would not be a good idea rather than the class' username method. It does work with (e.g.) `User($USER)` (my  env username). The problem is that to test this, I would need to create a new user that has that, and so far I have been failing.

Reply:
@MoSHad91 , @antarcticrainforest 
using the already existing `root` for testing as you suggested was the way. 
I will push and MR in another separate branch. thanks a lot